packages/client/tests/ecosystem/_utils/run.ts

```diff
@@ -0,0 +1,73 @@
+import fs from 'fs/promises'
+import os from 'os'
+import path from 'path'
+import { $ } from 'zx'
+
+async function main() {
+  console.log('🎠 Preparing ecosystem tests')
+  // we first get all the paths we are going to need to run ecosystem
+  const tmpDir = await fs.mkdtemp(path.join(os.tmpdir(), 'prisma-build'))
+  const cliPkgPath = path.join(__dirname, '..', '..', '..', '..', 'cli')
+  const clientPkgPath = path.join(__dirname, '..', '..', '..', '..', 'client')
+  const clientRuntimeDtsPath = path.join(clientPkgPath, 'runtime', 'index.d.ts')
+  const cliPkgJsonPath = path.join(cliPkgPath, 'package.json')
+  const clientPkgJsonPath = path.join(clientPkgPath, 'package.json')
+  const cliPkgJson = require(cliPkgJsonPath)
+  const clientPkgJson = require(clientPkgJsonPath)
+
+  // this process will need to modify some package.json, we save copies
+  await $`cd ${tmpDir} && cp ${cliPkgJsonPath} cli.package.json`.quiet()
+  await $`cd ${tmpDir} && cp ${clientPkgJsonPath} client.package.json`.quiet()
+  await $`cd ${tmpDir} && cp ${clientRuntimeDtsPath} client.runtime.d.ts`.quiet()
+
+  // we provide a function that can revert modified package.json back
+  const restoreOriginal = async () => {
+    await $`cd ${tmpDir} && cp cli.package.json ${cliPkgJsonPath}`.quiet()
+    await $`cd ${tmpDir} && cp client.package.json ${clientPkgJsonPath}`.quiet()
+    await $`cd ${tmpDir} && cp client.runtime.d.ts ${clientRuntimeDtsPath}`.quiet()
+  }
+
+  // if process is killed by hand, ensure that package.json is restored
+  process.on('SIGINT', () => restoreOriginal().then(() => process.exit(0)))
+
+  // use bundleDependencies to directly include deps like @prisma/engines
+  cliPkgJson.bundleDependencies = Object.keys(cliPkgJson.dependencies)
+  clientPkgJson.bundleDependencies = Object.keys(clientPkgJson.dependencies)
+
+  // write the modified package.json to overwrite the original package.json
+  await fs.writeFile(cliPkgJsonPath, JSON.stringify(cliPkgJson, null, 2))
+  await fs.writeFile(clientPkgJsonPath, JSON.stringify(clientPkgJson, null, 2))
+  // this is to avoid bundling types and locally link directly to the sources
+  await fs.writeFile(clientRuntimeDtsPath, `export * from '/client/src/runtime/index'`)
+  // TODO: have a fast mode (this one) and a full mode with type bundling
```

Comment:
As mentioned before, not having a full build is nice to quickly test (ie. skipping type bundling here), but in a v2 we may want to enable a full build to test the package as it is published. Type bundling should produce compatible types, but could as well fail and we wouldn't notice that in ecosytem tests. So that's a future thing to enable.